gcloud/storage/blob.py

```diff
@@ -19,9 +19,10 @@
 import os
 import time
 import datetime
-from StringIO import StringIO
-import urllib
+from io import BytesIO
 
+import six
+from six.moves.urllib.parse import quote  # pylint: disable=F0401
```

Comment:
I need to monkey-patch `HTTPConnecdtion` in tests, which makes using the unqualified name lots trickier.